apps/smtp/src/modules/smtp/configuration/smtp-metadata-manager.ts

```diff
@@ -1,8 +1,19 @@
 import { SettingsManager } from "@saleor/app-sdk/settings-manager";
-import { SmtpConfig } from "./smtp-config-schema";
-import { fromAsyncThrowable, fromPromise, fromThrowable, ok, ResultAsync } from "neverthrow";
+import { fromPromise, fromThrowable, ok, ResultAsync } from "neverthrow";
+
 import { BaseError } from "../../../errors";
+import { racePromise } from "../../../lib/race-promise";
 import { createLogger } from "../../../logger";
+import { SmtpConfig } from "./smtp-config-schema";
+
+/*
+ *In case that pulling metadata takes too loong, the app will stuck and we won't know what is happening, since
+ *function that process lives only 25s. We've set timeout, so if pulling metadata takes too long, we will throw an error.
+ *
+ *Maximum that call can take up to 1.5s, so we've set timeout to 3s as a safe margin,
+ *anything longer should be considered as SLA violation
```

Comment:
We don't have to write about SLA. SLA is part of Saleor Cloud, while this is open source. Its enough to mention that this call will just fail to execute